### DISCUSSION TITLE
Too many database queries

### DISCUSSION THREAD

**Discussion by jdbit** [Radi85/Comment#256] [Q&A]
Hi!
I like the app but it makes too many SQL queries which is not good when you have hundreds of comments, especially with pagination disabled. For example, with 20 comments it generates ~300 queries. It seems like the app requests user, user group, followers, and reactions for each comment as a separate SQL query. Is there any way to optimize it?

**Comment by Radi85** [comment]
Hi @jdbit 
Thanks for the feedback. I believe the sql queries can be optimized using Django Prefetch. Would you like to take a part and work on this task?

**Comment by jdbit** [comment]
Hi @Radi85
Thanks for your response. As far as I understand, prefetch_related does a separate lookup for each relationship and does the joining in Python, so it will not reduce the number of SQL queries. select_related (which does reduce the number of queries) can be used only to join the user table I guess. Though I'm not an experienced developer, I'll see what I can do.
